dapp/client/App.elm

```diff
@@ -67,19 +73,33 @@ update : Message -> Model -> ( Model, Cmd Message )
 update message model =
     case message of
         AddressChanged newAddress ->
-            ( { model | address = Just newAddress }, Cmd.none )
+            ( { initialModel | address = Just newAddress }, Cmd.none )
+
+        WalletChanged newWallet ->
+            ( { model | wallet = Just newWallet, eth = Nothing }, Cmd.none )
 
         RefreshAddress ->
             ( model, requestAddress () )
 
+        RequestWallet ->
+            case model.address of
+                Nothing ->
+                    ( model, Cmd.none )
```

Comment:
If `address` is not set, then requesting a wallet is essentially a no-op. Our view code doesn't allow for this state, but because address is a Maybe, we must handle the case when it is not set. This is how Elm prevents you from ever getting null or undefined errors.